go/private/rules/nogo.bzl

```diff
@@ -47,9 +46,8 @@ def _nogo_impl(ctx):
     if ctx.attr.debug:
         nogo_args.add("-debug")
     nogo_inputs = []
-    analyzer_archives = [get_archive(dep) for dep in ctx.attr.deps]
-    analyzer_importpaths = [archive.data.importpath for archive in analyzer_archives]
-    nogo_args.add_all(analyzer_importpaths, before_each = "-analyzer_importpath")
+    for dep in ctx.attr.deps:
+        nogo_args.add("-analyzer_importpath", dep[GoArchive].data.importpath)
```

Comment:
Is this really an improvement memory-wise?

**Old version**
* reference to a vector arg
* reference to `"-analyzer_importpath"`
* number of import paths
* for each import path
  * reference to import path

`12 + 4*N` bytes

**New version**
* for each import path
  * reference to `"-analyzer_importpath"`
  * reference to import path

`8*N` bytes

Reply:
Doesn't the old version also have an extra `8 + 4*N` bytes for the `analyzer_importpaths` list as well? Or does that not matter because it's not retained?

Reply:
It's not retained. `add_all` drains all lists (but not depsets) into its internal array. This makes it more efficient than related `add` calls (unless you use none of the keyword args).

Reply:
ah ok, I tried one more variant with the `map_each` wdyt?

Reply:
The additional memory usage is constant, but I don't see this part as an improvement to code structure or readability. `map_each` should be used only when it avoids retaining certain objects, but the import paths are already retained. How about just inlining the `get_archive` call?

Reply:
ah, I thought it would push work out of the analysis phase. I'll just simplify it in that case